central/complianceoperator/v2/scanconfigurations/datastore/datastore_impl.go

```diff
@@ -52,6 +52,58 @@ func (ds *datastoreImpl) ScanConfigurationExists(ctx context.Context, scanName s
 	return len(scanConfigs) > 0, nil
 }
 
+// ScanConfigurationProfileExists takes all the profiles being referenced by the scan configuration and checks if any cluster is using it in any existing scan configurations.
+func (ds *datastoreImpl) ScanConfigurationProfileExists(ctx context.Context, profiles []string, clusters []string) (bool, error) {
+	// Retrieve all scan configurations for the specified clusters.
+	scanConfigs, err := ds.storage.GetByQuery(ctx, search.NewQueryBuilder().
+		AddExactMatches(search.ClusterID, clusters...).ProtoQuery())
+	if err != nil {
+		return false, err
+	}
+
+	// Create a map for quick lookup of profiles.
+	profileMap := make(map[string]bool)
+
+	for _, scanConfig := range scanConfigs {
```

Comment:
Thinking ahead to the update case, we need to exclude the scan configuration we are trying to add or update.  Because if we have an update, then this would likely match to itself and prevent the update.